### PR TITLE
OSDOCS 6914: Updating pod examples to comply with restricted PSA (Nod…

### PR DESCRIPTION
**Pull request by bergerhoffer**
…es book)

Version(s):
4.15+

Issue:
https://issues.redhat.com/browse/OSDOCS-6914

Link to docs preview:
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/clusters/nodes-cluster-resource-configure.html#nodes-cluster-resource-configure-request-limit_nodes-cluster-resource-configure
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/clusters/nodes-cluster-resource-levels.html#nodes-cluster-resource-levels-command_nodes-cluster-resource-levels
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/clusters/nodes-cluster-resource-levels.html#nodes-cluster-resource-levels-job_nodes-cluster-resource-levels
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/containers/nodes-containers-downward-api.html#nodes-containers-downward-api-container-configmaps_nodes-containers-downward-api
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/containers/nodes-containers-downward-api.html#nodes-containers-downward-api-container-envars_nodes-containers-downward-api
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/containers/nodes-containers-downward-api.html#nodes-containers-downward-api-container-escaping_nodes-containers-downward-api
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/containers/nodes-containers-downward-api.html#nodes-containers-downward-api-container-secrets_nodes-containers-downward-api
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/containers/nodes-containers-downward-api.html#nodes-containers-downward-api-container-values-envars_nodes-containers-downward-api
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/containers/nodes-containers-downward-api.html#nodes-containers-downward-api-container-values-plugin_nodes-containers-downward-api
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/containers/nodes-containers-init.html#nodes-containers-init-creating_nodes-containers-init
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/containers/nodes-containers-projected-volumes.html#projected-volumes-examples_nodes-containers-projected-volumes
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/containers/nodes-containers-projected-volumes.html#nodes-containers-projected-volumes-creating_nodes-containers-projected-volumes
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/containers/nodes-containers-volumes.html#nodes-containers-volumes-subpath_nodes-containers-volumes
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/pods/nodes-pods-configmaps.html#nodes-pods-configmaps-use-case-consuming-in-env-vars_configmaps
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/pods/nodes-pods-configmaps.html#nodes-pods-configmaps-use-case-consuming-in-volumes_configmaps
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/pods/nodes-pods-configmaps.html#nodes-pods-configmaps-use-case-setting-command-line-arguments_configmaps
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/pods/nodes-pods-configuring.html#nodes-pods-configuring-critical_nodes-pods-configuring
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/pods/nodes-pods-priority.html#nodes-pods-priority-configuring_nodes-pods-priority
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/pods/nodes-pods-secrets.html#nodes-pods-secrets-certificates-creating_nodes-pods-secrets
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/pods/nodes-pods-secrets.html#nodes-pods-secrets-creating_nodes-pods-secrets
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/pods/nodes-pods-using.html#nodes-pods-using-example_nodes-pods-using-ssy
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/scheduling/nodes-scheduler-node-affinity.html#nodes-scheduler-node-affinity-about_nodes-scheduler-node-affinity
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/scheduling/nodes-scheduler-node-affinity.html#nodes-scheduler-node-affinity-example_nodes-scheduler-node-affinity
* No preview for nodes-scheduler-node-names-configuring.adoc
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/scheduling/nodes-scheduler-node-selectors.html#nodes-scheduler-node-selectors-about_nodes-scheduler-node-selectors
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/scheduling/nodes-scheduler-pod-affinity.html#nodes-scheduler-pod-affinity-about_nodes-scheduler-pod-affinity
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/scheduling/nodes-scheduler-pod-affinity.html#nodes-scheduler-pod-affinity-configuring_nodes-scheduler-pod-affinity
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/scheduling/nodes-scheduler-pod-affinity.html#nodes-scheduler-pod-affinity-example_nodes-scheduler-pod-affinity
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/scheduling/nodes-scheduler-pod-affinity.html#nodes-scheduler-pod-anti-affinity-configuring_nodes-scheduler-pod-affinity
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/scheduling/nodes-scheduler-pod-topology-spread-constraints.html#nodes-scheduler-pod-topology-spread-constraints-configuring_nodes-scheduler-pod-topology-spread-constraints
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/scheduling/nodes-scheduler-pod-topology-spread-constraints.html#nodes-scheduler-pod-topology-spread-constraints-examples_nodes-scheduler-pod-topology-spread-constraints
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/scheduling/secondary_scheduler/nodes-secondary-scheduler-configuring.html#nodes-secondary-scheduler-pod-console_secondary-scheduler-configuring
* https://62602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/pods/run_once_duration_override/run-once-duration-override-install.html#rodoo-enable-override_run-once-duration-override-install

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by bergerhoffer** [comment]
FYI @mburke5678 Here's the PR I was talking about

File: modules/nodes-containers-init-creating.adoc [hunk 1]
```diff
@@ -21,14 +21,23 @@ metadata:
...
     command: ['sh', '-c', 'echo The app is running! && sleep 3600']
+    securityContext:
+      allowPrivilegeEscalation: false
+      capabilities:
+        drop: [ALL]
   initContainers:
   - name: init-myservice
     image: registry.access.redhat.com/ubi9/ubi:latest
     command: ['sh', '-c', 'until getent hosts myservice; do echo waiting for myservice; sleep 2; done;']
+    # TODO: Do these containers need securityContext too?
```
Comment (bergerhoffer) [inline modules/nodes-containers-init-creating.adoc:40]: @s-urbaniak @stlaz Do containers under `initContainers` need securityContext set too?

File: modules/nodes-containers-sysctls-unsafe.adoc [hunk 2]
```diff
@@ -102,6 +102,7 @@ spec:
   - name: podexample
     image: centos
     command: ["bin/bash", "-c", "sleep INF"]
+    # TODO: How to update this example, if at all?
```
Comment (bergerhoffer) [inline modules/nodes-containers-sysctls-unsafe.adoc:105]: @s-urbaniak @stlaz This already has securityContext set - not sure if the existing settings are important to the section shown (unsafe sysctls)

File: modules/nodes-pods-using-example.adoc [hunk 3]
```diff
@@ -36,10 +36,11 @@ spec:
   schedulerName: default-scheduler
   terminationGracePeriodSeconds: 30
   nodeName: ip-10-0-140-16.us-east-2.compute.internal
+  # TODO: How to update this example? Should we just replace it?
```
Comment (bergerhoffer) [inline modules/nodes-pods-using-example.adoc:39]: @s-urbaniak @stlaz  This one has securityContext set too, but callout <3> below mentions about it being restrictive. Can we just replace the existing one with what we've been updating, or do we need to do something different here?

File: modules/rodoo-enable-override.adoc [hunk 4]
```diff
@@ -40,16 +40,17 @@ metadata:
   namespace: <namespace>                 <1>
 spec:
   restartPolicy: Never                   <2>
+  securityContext:
+    runAsNonRoot: true
+    seccompProfile:
+      type: RuntimeDefault
   containers:
     - name: busybox
+    # TODO: This already had securityContext here on the container, but it was a mix. Updated, does this look okay?
```
Comment (bergerhoffer) [inline modules/rodoo-enable-override.adoc:49]: @s-urbaniak @stlaz I updated this one to what we've been adding. Does this look okay?

File: modules/nodes-containers-init-creating.adoc (hunk 1 above)
Comment (stlaz) [reply modules/nodes-containers-init-creating.adoc:40]: yes

File: modules/nodes-containers-sysctls-setting.adoc [hunk 5]
```diff
@@ -32,15 +32,16 @@ spec:
   - name: podexample
     image: centos
     command: ["bin/bash", "-c", "sleep INF"]
-    securityContext: 
-      runAsUser: 2000 
-      runAsGroup: 3000 
-      allowPrivilegeEscalation: false 
-      capabilities: 
+    # How to update this example?
```
Comment (stlaz) [inline modules/nodes-containers-sysctls-setting.adoc:35]: leave as is

File: modules/nodes-containers-sysctls-unsafe.adoc (hunk 2 above)
Comment (stlaz) [reply modules/nodes-containers-sysctls-unsafe.adoc:105]: leave as is

File: modules/nodes-pods-using-example.adoc (hunk 3 above)
Comment (stlaz) [reply modules/nodes-pods-using-example.adoc:39]: we should actually move it outside the "default" NS, too. This appears to be some kind of an ancient dump of a pod from the cluster. I don't think pods in the "default" NS would even get SCC annotations.

We should ask the owners  of the doc about what it's supposed to do with this one I guess.

File: modules/rodoo-enable-override.adoc (hunk 4 above)
Comment (stlaz) [reply modules/rodoo-enable-override.adoc:49]: the changes to this manifest look good to me

File: modules/nodes-containers-init-creating.adoc (hunk 1 above)
Comment (bergerhoffer) [reply modules/nodes-containers-init-creating.adoc:40]: Thanks, added!

File: modules/nodes-pods-using-example.adoc (hunk 3 above)
Comment (bergerhoffer) [reply modules/nodes-pods-using-example.adoc:39]: Thanks, I'll see who I can find to help with this one

Comment (bergerhoffer) [reply modules/nodes-pods-using-example.adoc:39]: @mburke5678 @sunilcio Do you know who could help figure out what we should do with this example?

This PR is in the context of trying to update all of our pod examples to comply with restricted PSA enforcement.

**Comment by kalexand-rh** [comment]
The `branch/enterprise-4.15` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.14. And any PR going into `main` must also target the latest version branch (enterprise-4.15).

If the update in your PR does NOT apply to version 4.15 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main

**Comment by kalexand-rh** [comment]
The `branch/enterprise-4.16` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.15. And any PR going into `main` must also target the latest version branch (enterprise-4.16).

If the update in your PR does NOT apply to version 4.16 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

**Comment by bergerhoffer** [comment]
Hi @sunilcio, I'm coming back to this PR on updating the pod examples in the Nodes book to comply with pod security admission restricted enforcement. 

I see that you gave it a lgtm, so I'm not sure if it all of the examples work properly that you've tested, but there is still one open conversation of an example that we're not sure how to update. Can you please take a look? https://github.com/openshift/openshift-docs/pull/62602/files#r1268646879

File: modules/nodes-pods-using-example.adoc (hunk 3 above)
Comment (sunilcio) [reply modules/nodes-pods-using-example.adoc:39]: This output looks dump of `oc describe pod` and not an an actual pod yaml file. Here we see output like `status`, `hostIP` which are not part of pod yaml. I see this section is intended to describe example pod yaml, I guess we can use pod example file from other section and add here and remove the current one?

File: modules/nodes-pods-using-example.adoc [hunk 6]
```diff
@@ -18,103 +18,52 @@ kind: Pod
...
-<9> The pod defines storage volumes that are available to its container(s) to use. In this case, it provides an ephemeral volume for a `secret` volume containing the default service account tokens. 
+<8> The pod defines storage volumes that are available to its container(s) to use.
 +
-If you attach persistent volumes that have high file counts to pods, those pods can fail or can take a long time to start. For
-more information, see link:https://access.redhat.com/solutions/6221251[When using Persistent Volumes with high file counts in OpenShift, why do pods fail to start or take an excessive amount of time to achieve "Ready" state?].
+If you attach persistent volumes that have high file counts to pods, those pods can fail or can take a long time to start. For more information, see link:https://access.redhat.com/solutions/6221251[When using Persistent Volumes with high file counts in OpenShift, why do pods fail to start or take an excessive amount of time to achieve "Ready" state?].
```
Comment (skopacz1) [inline modules/nodes-pods-using-example.adoc:66]: Doesn't seem like you wrote this line yourself, but just an FYI that this link takes you to an article with a different title:
```suggestion
If you attach persistent volumes that have high file counts to pods, those pods can fail or can take a long time to start. For more information, see link:https://access.redhat.com/solutions/6221251[Pods using Persistent Volumes with high file counts fail to start or take an excessive amount of time in OpenShift].
```

**Review by skopacz1 — COMMENTED** [review]
Left one nit for a line that's technically out of scope for the PR, otherwise I'm a bit terrified to give the rest of this XL PR a LGTM haha. Thanks a _lot_ for providing preview links respective to the diff order.

I'm still relatively new to reviewing, so please wait until @adellape approves my review before proceeding to merge.

**Review by adellape — COMMENTED** [review]
@skopacz1 Peer review LGTM 👍

**Comment by stlaz** [comment]
LGTM

**Pull request merged**